packages/record/src/trial.ts

```diff
@@ -32,7 +32,7 @@ export default class TrialRecordExtension implements JsPsychExtension {
 
   /** Ran when the trial has loaded. */
   public on_load() {
-    this.recorder?.start("trial_video");
+    this.recorder?.start(false, `${this.jsPsych.getCurrentTrial().type}`);
   }
```

Comment:
Argh, you're right, this is a class name, not the 'trial_type' string 🤦‍♀️  Fixing it now.

Reply:
You can get the current plugin name like this: `this.jsPsych.getCurrentTrial().type.info.name`, but for some reason the jsPsych types produce errors in that line (`info` property is missing from the jsPsych plugin class interface). So I created a new interface that extends the jsPsych plugin interface. 

I doubled checked in the jsPsych code that plugin classes must be created with PluginInfo, and PluginInfo must have a name property (string). Also, when running a trial, the jsPsych code assumes that a trial's `pluginInfo` value contains a `name`. So if a plugin class doesn't have a value for `info.name`, then I don't think it would work. 

Thus I don't think we need to do any error catching here for the existence of `info.name` on the class, but I'm open to suggestions if you disagree. If so, maybe we could put in a default value for the plugin name, or throw an error?

Reply:
Would the fact that `TrialRecordExtension` is not a plugin change anything?

Reply:
Nope, the fact that this is an extension is the reason why getting the plugin type here is tricky. The `TrialRecordExtension` can run over any plugin, and the plugin info isn't automatically accessible from within the extension code. But since we're waiting for the plugin to load before we start recording, we can also ask jsPsych at that point what the current trial type is (`this.jsPsych.getCurrentTrial().type`). This gives us the name of the plugin class (e.g. `HtmlKeyboardResponsePlugin`), and from that we can use `.info.name` to get the plugin/trial type string (e.g. "html-keyboard-response").